db/error_handler.cc

```diff
@@ -268,10 +270,13 @@ Status ErrorHandler::SetBGError(const IOStatus& bg_io_err,
     Status bg_err(new_bg_io_err, Status::Severity::kHardError);
     EventHelpers::NotifyOnBackgroundError(db_options_.listeners, reason, &s,
                                           db_mutex_, &auto_recovery);
+    if (recovery_in_prog_ && recovery_error_.ok()) {
+      recovery_error_ = bg_err;
+    }
     if (bg_err.severity() > bg_error_.severity()) {
```

Comment:
@anand1976 Yes. In the testing, the auto resume is actually not the one that reschedule the compaction. The compaction thread reschedules it.

Reply:
@anand1976 The `bg_error_` is set to Status::Severity::kHardError if the previous one is a Retryable error. Do you mean we need to test `BackgroundErrorReason reason` and set to `kSoftError` for compaction case here?

Reply:
Since a compaction failure doesn't immediately affect future writes to the DB, the error severity can be set to ```kSoftError```. Ideally, we can do the same even for flush errors if WAL is not in use, so a user that doesn't use WAL will see no impact due to retryable errors. It can be done in a follow-on PR if that makes it easier.

Reply:
@anand1976 Sure, will create a task to track it and do it in a follow-on PR.